src/test/javascript/spec/service/exam-monitoring.service.spec.ts

```diff
@@ -0,0 +1,168 @@
+import { ArtemisServerDateService } from 'app/shared/server-date.service';
+import { StudentExam } from 'app/entities/student-exam.model';
+import { HttpResponse } from '@angular/common/http';
+import { of } from 'rxjs';
+import {
+    ConnectionUpdatedAction,
+    ContinuedAfterHandedInEarlyAction,
+    EndedExamAction,
+    ExamAction,
+    ExamActionType,
+    ExamActivity,
+    HandedInEarlyAction,
+    SavedExerciseAction,
+    StartedExamAction,
+    SwitchedExerciseAction,
+} from 'app/entities/exam-user-activity.model';
+import { ExamMonitoringService } from 'app/exam/monitor/exam-monitoring.service';
+
+describe('ExamMonitoringService', () => {
+    let service: ExamMonitoringService;
+    let exam: Exam;
+    let studentExam: StudentExam;
+    let course: Course;
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            imports: [ArtemisTestModule],
+            providers: [ArtemisServerDateService, ExamMonitoringService],
+        })
+            .compileComponents()
+            .then(() => {
+                service = TestBed.inject(ExamMonitoringService);
+            });
+    });
+
+    beforeEach(() => {
+        // reset exam
+        exam = new Exam();
+        exam.id = 1;
+        exam.monitoring = true;
+
+        // reset studentExam
+        studentExam = new StudentExam();
+        studentExam.id = 1;
+
+        // reset course
+        course = new Course();
+        course.id = 1;
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+
+    it('should handle exam actions when monitoring enabled', () => {
+        const examActions = createExamActions();
+        expect(examActions.length).toBe(Object.keys(ExamActionType).length);
+        examActions.forEach((action) => {
+            service.handleActionEvent(studentExam, action, exam.monitoring!);
+            expect(studentExam.examActivity).not.toBeUndefined();
+            expect(studentExam.examActivity!.examActions).not.toBeUndefined();
+            expect(studentExam.examActivity!.examActions.length).toBe(1);
+            expect(studentExam.examActivity!.examActions.first()).toBe(action);
+            studentExam.examActivity!.examActions = [];
+        });
+    });
+
+    it('should not handle exam actions when monitoring disabled', () => {
+        exam.monitoring = false;
+        const examActions = createExamActions();
+        expect(examActions.length).toBe(Object.keys(ExamActionType).length);
+        examActions.forEach((action) => {
+            service.handleActionEvent(studentExam, action, exam.monitoring!);
+            expect(studentExam.examActivity).toBeUndefined();
+        });
+    });
+
+    it('should sync actions when monitoring enabled', () => {
+        const spy = jest.spyOn(service, 'syncActions').mockReturnValue(httpResponse());
+        studentExam.examActivity = new ExamActivity();
+        service.saveActions(exam, studentExam, course.id!);
+        expect(spy).toHaveBeenCalled();
```

Comment:
I think you forgot this one :D
```suggestion
        expect(spy).toHaveBeenCalledOnce();
```